apple-codesign/src/windows.rs

```
@@ -0,0 +1,736 @@
+    }
+}
+
+fn find_certificates(
+    store_name: StoreName,
+    store_type: StoreType,
+) -> Result<Vec<StoreCertificate>, AppleCodesignError> {
+    let mut certs = vec![];
+
+    let store_type_as_str = <StoreType as std::convert::Into<&'static str>>::into(store_type);
+    let store_name_as_str = <StoreName as std::convert::Into<&'static str>>::into(store_name);
+
+    let store_type_as_wstr = U16CString::from_str(store_type_as_str);
+    if store_type_as_wstr.is_err() {
+        return Err(AppleCodesignError::WindowsStoreError(format!("could not convert store type {} to wide string (this should not happen)", store_type_as_str)));
+    }
+
+    let dwflags = CERT_OPEN_STORE_FLAGS::from(store_name);
+
+    // Open the certificate store.
+    let store_handle = unsafe {
+        CertOpenStore(
+            CERT_STORE_PROV_SYSTEM_W as _,
+            X509_ASN_ENCODING | PKCS_7_ASN_ENCODING,
+            HCRYPTPROV_LEGACY::default(),
+            dwflags | CERT_STORE_OPEN_EXISTING_FLAG,
+            store_type_as_wstr.unwrap().as_ptr() as _,
+        )
+    };
+    if store_handle.is_null() {
+        return Err(AppleCodesignError::WindowsStoreError(format!("could not open store {} with store type {} (this should not happen): 0x{:08X}", store_name_as_str, store_type_as_str, get_last_error())));
+    }
+
+    // Enumerate the certificates.
+    let mut cert_context = ptr::null_mut();
+    loop {
+        // Get the next certificate.
+        cert_context = unsafe {
+            CertFindCertificateInStore(
+                store_handle,
+                X509_ASN_ENCODING | PKCS_7_ASN_ENCODING,
+                0,
+                CERT_FIND_ANY,
+                ptr::null_mut(),
+                cert_context,
+            )
+        };
+        if cert_context.is_null() {
+            break;
+        }
+        let cert = match StoreCertificate::new(cert_context) {
+            Ok(cert) => cert,
+            Err(()) => continue,
+        };
```

Silently ignoring errors here feels like it could be a source of confusion for end-users. Depending on how frequently this occurs, I'd prefer at least an `info!()` log message or `warn!()` if we feel it will be uncommon with some details on what's going on.

---

Added logging with `error!()`. Let me know if you want me to tone it down.